single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultLabelProvider.java

```diff
@@ -179,25 +163,22 @@ public List<ElectricalNodeInfo> getElectricalNodesInfos(VoltageLevelGraph graph)
                 .collect(Collectors.toList());
     }
 
-    private void addBranchStatusDecorator(List<NodeDecorator> nodeDecorators, Node node, Direction direction, Connectable<?> c) {
-        OperatingStatus<?> branchStatus = (OperatingStatus<?>) c.getExtension(OperatingStatus.class);
-        if (branchStatus != null) {
-            switch (branchStatus.getStatus()) {
-                case PLANNED_OUTAGE:
-                    nodeDecorators.add(getBranchStatusDecorator(node, direction, PLANNED_OUTAGE_BRANCH_NODE_DECORATOR));
-                    break;
-                case FORCED_OUTAGE:
-                    nodeDecorators.add(getBranchStatusDecorator(node, direction, FORCED_OUTAGE_BRANCH_NODE_DECORATOR));
-                    break;
-                default:
-                    break;
+    private void addBranchStatusDecorator(List<NodeDecorator> nodeDecorators, Node node, Direction direction, Identifiable<?> identifiable) {
+        if (identifiable != null) {
+            OperatingStatus<?> branchStatus = (OperatingStatus<?>) identifiable.getExtension(OperatingStatus.class);
```

Comment:
Doesn't something like this avoid the code smell? Not so sure...
```suggestion
    private <T extends Identifiable<T>> void addBranchStatusDecorator(List<NodeDecorator> nodeDecorators, Node node, Direction direction, Identifiable<T> identifiable) {
        if (identifiable != null) {
            OperatingStatus<T> branchStatus = identifiable.getExtension(OperatingStatus.class);
```